### ISSUE TITLE
Running tests that depend on string[] TypeDescriptor converters fail.

### ISSUE THREAD

**Issue by NTaylorMullen** [Microsoft/vstest#427]
## Description
In MVC there's a `ModelMetadata` system that inspects types and tracks certain information about said types. One of the things it tracks is `IsComplexType`. It computes this by doing: `!TypeDescriptor.GetConverter(ModelType).CanConvertFrom(typeof(string));`, [code for ref](https://github.com/aspnet/Mvc/blob/4c62eb89af1a5b7890499c5afe240095b6c2db30/src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelMetadata.cs#L430). This works for most scenarios but when `ModelType` is `string[]` `IsComplexType` ends up returning `false` when it should be true.

Looking through the vstest code it looks like it's due to how you guys register your converter for [string arrays](https://github.com/Microsoft/vstest/blob/c701070abb126853b58800cedd7554482471ac7d/src/Microsoft.TestPlatform.ObjectModel/TestObject.cs#L27). The [convert](https://github.com/Microsoft/vstest/blob/c701070abb126853b58800cedd7554482471ac7d/src/Microsoft.TestPlatform.ObjectModel/TestProperty/CustomStringArrayConverter.cs#L17) is a little too simple to handle string arrays properly.

This breaks the following MVC tests:
- https://github.com/aspnet/Mvc/blob/760c8f38678118734399c58c2dac981ea6e47046/test/Microsoft.AspNetCore.Mvc.IntegrationTests/KeyValuePairModelBinderIntegrationTest.cs#L503
- https://github.com/aspnet/Mvc/blob/760c8f38678118734399c58c2dac981ea6e47046/test/Microsoft.AspNetCore.Mvc.IntegrationTests/SimpleTypeModelBinderIntegrationTest.cs#L496  

Working around this for now by skipping the test in core. This somehow works in net451 tests because the `GetConverter` call returns an array converter instead of the above mentioned [CustomStringArrayConverter](https://github.com/Microsoft/vstest/blob/c701070abb126853b58800cedd7554482471ac7d/src/Microsoft.TestPlatform.ObjectModel/TestProperty/CustomStringArrayConverter.cs#L17).

## Steps to reproduce
- Download [TypeDescriptorBug.zip](https://github.com/Microsoft/vstest/files/745880/TypeDescriptorBug.zip)
- cd TypeDescriptorBug
- dotnet restore
- dotnet test

## Expected behavior
- Tests pass

## Actual behavior
- Tests fail

## Environment
```
.NET Command Line Tools (1.0.0-rc4-004616)

Product Information:
 Version:            1.0.0-rc4-004616
 Commit SHA-1 hash:  56d8071361

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.14393
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Users\nimullen\AppData\Local\Microsoft\dotnet\sdk\1.0.0-rc4-004616
```

**Comment by codito** [comment]
Triage: this issue is a core part of test platform, but there isn't a workaround. We should investigate the fix for this (may be for RTM).

**Comment by codito** [comment]
Fixed with b78eafa  in master, bd04f21d5ce9e7474ae2d3f178416d67ab1d94d6 in 15.0-rtm.
